psi4/driver/driver_cbs.py

```diff
@@ -25,34 +25,152 @@
 #
 # @END LICENSE
 #
+"""Plan, run, and assemble QC tasks to obtain composite method, basis, & options treatments.
+
+===
+CBS
+===
+
+cbs_text_parser()
+-----------------
+
+    _parse_cbs_gufunc_string()
+    --------------------------
+    * break user string into paired method and basis stages
+
+* transform user string into cbs kwargs inc'l basic cbs_metadata
+
+
+----------------------------
+CompositeComputer.__init__()
+----------------------------
+
+    _process_cbs_kwargs()
+    ---------------------
+    * transform user kwargs into trial cbs_metadata format (aka dict spec)
+
+        _validate_cbs_inputs()
+        ----------------------
+
+            _get_default_xtpl()
+            -------------------
+            * supply default xtpl fn for stage and basis conditions
+
+            _expand_bracketed_basis()
+            -------------------------
+            * parse and validate user bases
+
+        * check and supply defaults for cbs_metadata format (various calls to above two fns)
+
+* BaseComputer.__init__()
+
+    _build_cbs_compute()
+    --------------------
+
+        _expand_scheme_orders()
+        -----------------------
+        * form f_fields dict of entries for each zeta in a scheme (single NEED; entries related by nonlinear fn)
+
+        _contract_bracketed_basis()
+        ---------------------------
+        * form basis abbr. string from basis seq
+
+    * form d_fields list of stages or stage halves from NEEDs (GRAND_NEED; items related linearly to form final val)
```

Comment:
What's a stage?

Reply:
stages are the usual decomposition of a composite method -- scf part, correlated part, deltas. I've added a definition.